### PR TITLE
Add handling for refund failure in payment timeline

### LINKED ISSUE #1476: Add refund failures info to transaction timeline
Refunds failures are not displayed on the transaction timeline.
Although we have order notes on the order page, it's based on webhooks forwarding and unreliable sometimes. This leads to confusing look of transaction timeline displaying a successful refund:

image (6)

This need to be fixed to match. Here is the timeline on Stripe dashboard for reference:

image (7)

Initial discussion and a bit more context (regarding ARN) is in slack (p1617090705365000-slack-CGGCLBN58)

### PR DESCRIPTION
**Pull request by deepakpathania**
Fixes #1476 

#### Changes proposed in this Pull Request
- Add handling for `refund_failed` event in the payment timeline by adding a new line item.
- Update the payment summary status to reflect the refund failure.

Screenshot for reference with the updated timeline:

Screenshot 2022-03-21 at 7 00 47 PM

#### Testing instructions

* Checkout this server PR: 1690-gh-Automattic/woocommerce-payments-server that introduces the new `refund_failed` timeline event.
* Follow the testing instructions there to get an event with refund failure.
* Verify that the new timeline event for refund failure is added.
* Verify that the status in the payment summary at the top is updated to `Refund failure`.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: client/payment-details/timeline/map-events.js [hunk 1]
```diff
@@ -311,6 +311,25 @@ const composeFXString = ( event ) => {
 	);
 };
 
+// Conditionally adds the ARN details to the timeline in case they're available.
+const getRefundTrackingDetails = ( event ) => {
+	return 'available' === event.acquirer_reference_number_status
+		? `Acquirer Reference Number(ARN) ${ event.acquirer_reference_number }`
```
Comment (htdat) [inline client/payment-details/timeline/map-events.js:317]: This text should use `__()` for the localization purpose. 

Also nitpick: there may be a space before `(ARN)`.

File: client/payment-details/timeline/map-events.js [hunk 2]
```diff
@@ -311,6 +311,25 @@ const composeFXString = ( event ) => {
...
+		: '';
+};
+
+// Converts the failure reason enums to error messages.
+const getRefundFailureReason = ( event ) => {
+	switch ( event.failure_reason ) {
+		case 'expired_or_canceled_card':
+			return 'the card being expired or canceled.';
```
Comment (htdat) [inline client/payment-details/timeline/map-events.js:325]: All returned strings here would use `__()` for the localization purpose too.

**Review by htdat — COMMENTED** [review]
✅  It's working well in my manual tests. Code looks good too.

💬  I have some comments related to string translations too. 

❓ ARN numbers are always helpful for merchants to communicate with their shoppers, not only in the case of failed refund. Can we create a follow-up issue to add these numbers for successful refunds in another PR?

File: client/payment-details/timeline/map-events.js (hunk 1 above)
Comment (deepakpathania) [reply client/payment-details/timeline/map-events.js]: Fixed

File: client/payment-details/timeline/map-events.js (hunk 2 above)
Comment (deepakpathania) [reply client/payment-details/timeline/map-events.js]: Good catch, added.

**Comment by deepakpathania** [comment]
> Can we create a follow-up issue to add these numbers for successful refunds in another PR?

Created an issue for adding ARN to successful refunds [here](https://github.com/Automattic/woocommerce-payments/issues/4024).

**Review by htdat — APPROVED** [review]
Thanks for addressing my feedback.

**Pull request merged**